modelci/finetuner/torch_data_module.py

```diff
@@ -3,120 +3,90 @@
-        self.target_transform = target_transform
-
-    def __getitem__(self, index: int) -> Tuple[object, int]:
-        """
-        Reference: https://github.com/pytorch/vision/blob/master/torchvision/datasets/cifar.py#L105-L128
-        Args:
-            index (int): Index
-        Returns:
-            tuple: (Tuple[object, int]) where target is index of the target class.
-        """
-        img, target = self.data[index], self.targets[index]
-
-        # doing this so that it is consistent with all other datasets
-        # to return a PIL Image
-        img = Image.fromarray(img)
+import torchvision
+from torch.utils.data import random_split, DataLoader
+from torchvision.transforms import transforms
 
-        if self.transform is not None:
-            img = self.transform(img)
+# transforms
+from modelci.finetuner import OUTPUT_DIR
 
-        if self.target_transform is not None:
-            target = self.target_transform(target)
+input_size = (224, 224)
 
-        return img, target
+train_transforms = transforms.Compose([
+    transforms.RandomResizedCrop(input_size),
+    transforms.RandomHorizontalFlip(),
+    transforms.ToTensor(),
+    transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
+])
 
-    def __len__(self) -> int:
-        return len(self.data)
+test_transforms = transforms.Compose([
+    transforms.Resize(input_size),
+    transforms.CenterCrop(input_size),
+    transforms.ToTensor(),
+    transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
+])
 
 
+# noinspection PyAbstractClass
 class PyTorchDataModule(pl.LightningDataModule):
 
-    def __init__(self, dataset_name: str, input_size: Tuple[int, int]=(224,224), batch_size: int=8):
+    def __init__(self, dataset_name: str, batch_size: int = 8, num_workers=2, data_dir=OUTPUT_DIR):
         """
 
         Args:
-            dataset_name (str): name of dataset to be load, full lists of supported datasets: https://www.tensorflow.org/datasets/catalog/overview#all_datasets
-            input_size (Tuple[int, int]): input image size of target model
+            dataset_name (str): name of vision dataset to be load, full lists of supported datasets:
+                https://pytorch.org/docs/stable/torchvision/datasets.html
             batch_size (int): samples per batch to load
         """
         super().__init__()
+        self.data_dir = data_dir
         self.dataset_name = dataset_name
-        self.input_size = input_size
         self.batch_size = batch_size
-        self.data_builder = tfds.builder(self.dataset_name)
+        self.num_workers = num_workers
 
-    def prepare_data(self):
-        self.data_builder.download_and_prepare()
+        if hasattr(torchvision.datasets, self.dataset_name):
+            self.dataset = getattr(torchvision.datasets, self.dataset_name)
+        else:
+            raise ValueError(f'torchvision.datasets does not have dataset name {self.dataset_name}')
```

Comment:
raise two errors

One error is the same as yours.
The other error is "You are using a customized dataset. Please define your dataset with the provided dataset_builder function."

Reply:
dataset_builder function should also contain a frontned @univerone

Reply:
dataset_type

imagenet
coco